synapse/handlers/e2e_keys.py

```diff
@@ -280,33 +280,25 @@ def get_cross_signing_keys_from_cache(self, query, from_user_id):
             defer.Deferred[dict[str, dict[str, dict]]]: map from
                 (master_keys|self_signing_keys|user_signing_keys) -> user_id -> key
         """
-        master_keys = {}
-        self_signing_keys = {}
         user_signing_keys = {}
 
-        for user_id in query:
-            # XXX: consider changing the store functions to allow querying
-            # multiple users simultaneously.
-            key = yield self.store.get_e2e_cross_signing_key(
-                user_id, "master", from_user_id
-            )
-            if key:
-                master_keys[user_id] = key
+        users = list(query)
 
-            key = yield self.store.get_e2e_cross_signing_key(
-                user_id, "self_signing", from_user_id
-            )
-            if key:
-                self_signing_keys[user_id] = key
+        master_keys = yield self.store.get_e2e_cross_signing_keys_bulk(
+            users, "master", from_user_id
+        )
+        self_signing_keys = yield self.store.get_e2e_cross_signing_keys_bulk(
+            users, "self_signing", from_user_id
+        )
```

Comment:
I'd be somewhat tempted to make these calls pull out *all* keys for a user, and then filter out the `user_signing` key

Reply:
If we add new types of keys in the future, we may or may not want them to be public.  It seems safer to just fetch the ones that we want.

Reply:
Ah right. Maybe we could filter for things we want then? Or combine the functions and have the SQL do `WHERE type IN (...)` or something?